pool.go

```diff
@@ -207,51 +228,98 @@ func (p *pool) fetchWith(ctx context.Context, c cid.Cid, with string) (blk block
 
 	for i := 0; i < len(nodes); i++ {
 		blk, err = p.doFetch(ctx, nodes[i], c)
-		if err != nil {
-			p.lk.RLock()
-			idx := -1
-			var nm *Member
-			var needUpdate bool
-			for j, m := range p.endpoints {
-				if m.String() == nodes[i] {
-					if nm, needUpdate = m.Downvote(p.config.PoolFailureDownvoteDebounce); needUpdate {
-						idx = j
+		var idx int
+		var nm *Member
+
+		if err == nil {
```

Comment:
We don't want to lose visibility introduced by https://github.com/filecoin-saturn/caboose/pull/24
```suggestion
		if err != nil {
				goLogger.Debugw("fetch failed", "from", nodes[i], "of", c, "attempt", i, "error", err)
		} else {
```

Reply:
Done.